src/qtt/algorithms/markov_chain.py

```diff
@@ -163,54 +177,57 @@ def stationary_distribution_discrete(jump_chain):
         pi = _solve_least_squares(A, B)
         return pi
 
-    def generate_sequence(self, length, delta_time, initial_state=None):
+    def generate_sequence(self, length: int, delta_time: float,
+                          initial_state: Union[None, int, np.ndarray] = None) -> np.ndarray:
         """ Generate a random sequence with the model
 
         Args:
-            length (int): number of elements in the sequence
-            delta_time (float): time step to be used. This is equal to one over the samplerate.
-            initial_state (None or int or list): This parameter determines how the first element of the generated
+            length: number of elements in the sequence
+            delta_time: time step to be used. This is equal to one over the samplerate.
+            initial_state: This parameter determines how the first element of the generated
                 sequence is chosen. If an int, then use that state is initial state. If None then take
                 a random state weighted by the stationary distribution. If the initial_state is a list, then the list
                 is interpreted as a probability distribution and the first element is samples from all possible states
                 according to the distribution specified.
         Returns:
-            array : generated sequence
+            Array with generated sequence
 
         """
         number_of_states = self.number_of_states()
         if initial_state is None:
-            initial_state = self.stationary_distribution()
-            initial_state = random.choices(range(number_of_states), weights=initial_state, k=1)[0]
+            initial_state = self.stationary_distribution().flatten().tolist()
+            initial_state = random.choices(range(number_of_states), weights=initial_state, k=1)[0]  # type: ignore
```

Comment:
is mypy happy with this reuse of initial_state?

Reply:
mypy is not happy because `random.choices` returns a list without type annotation. the input argument to `random.choices` is `range(number_of_states)` we we know it is an int